java/src/main/java/ai/rapids/cudf/ParquetColumnWriterOptions.java

```diff
@@ -239,68 +239,96 @@ public T withNullableTimestampColumn(String name, boolean isInt96) {
     this(columnName, true);
   }
 
+  @FunctionalInterface
+  protected interface ByteArrayProducer {
+    boolean[] apply(ParquetColumnWriterOptions opt);
+  }
+
+  @FunctionalInterface
+  protected interface IntArrayProducer {
+    int[] apply(ParquetColumnWriterOptions opt);
+  }
+
   boolean[] getFlatIsTimeTypeInt96() {
     boolean[] ret = {isTimestampTypeInt96};
+    return getFlatBooleans(ret, (opt) -> opt.getFlatIsTimeTypeInt96());
+  }
 
-    for (ParquetColumnWriterOptions opt: childColumnOptions) {
-      boolean[] b = opt.getFlatIsTimeTypeInt96();
-      boolean[] tmp = new boolean[ret.length + b.length];
-      System.arraycopy(ret, 0, tmp, 0, ret.length);
-      System.arraycopy(b, 0, tmp, ret.length, b.length);
-      ret = tmp;
+  protected boolean[] getFlatBooleans(boolean[] ret, ByteArrayProducer producer) {
+    boolean[][] childResults = new boolean[childColumnOptions.length][];
+    int totalChildrenFlatLength = 0;
```

Comment:
Rather than needing to remember to tack on `ret.length` later, we can initialize the total length to `ret.length` here.  Otherwise it's a bit weird that the variable name implies it's the total length when it's actually not.